src/main/java/com/aws/greengrass/util/FileSystemPermission.java

```diff
@@ -70,6 +82,90 @@
         return ret;
     }
 
+    /**
+     * Convert to a list of Acl entries for use with AclFileAttributeView.setAcl.
+     *
+     * @param path path to apply to
+     * @return List of Acl entries
+     * @throws IOException if any exception occurs while converting to Acl
+     */
+    public List<AclEntry> toAclEntries(Path path) throws IOException {
+        // These sets of permissions are reverse engineered by setting the permission on a file using Windows File
+        // Explorer. Then use AclFileAttributeView.getAcl to examine what were set.
+        final Set<AclEntryPermission> readPerms = new HashSet<>(Arrays.asList(
+                AclEntryPermission.READ_DATA,
+                AclEntryPermission.READ_NAMED_ATTRS,
+                AclEntryPermission.READ_ATTRIBUTES,
+                AclEntryPermission.READ_ACL,
+                AclEntryPermission.SYNCHRONIZE));
+        final Set<AclEntryPermission> writePerms = new HashSet<>(Arrays.asList(
```

Comment:
Is this list of acls the entire list? I feel like windows had many many more options, including specific options for files vs. directories.

Reply:
This is not the entire list of available permissions. `writePerms` is the list of permissions that would make File Explorer UI (right click on the file->Properties->Security->username) shows the user has write permission.

The complete list of permission is here:
https://docs.oracle.com/javase/8/docs/api/java/nio/file/attribute/AclEntryPermission.html

See also:
https://docs.oracle.com/javase/8/docs/api/java/nio/file/attribute/AclEntry.html

Reply:
write needs DELETE and DELETE_CHILD

Reply:
what about WRITE_ACL and WRITE_OWNER?

Reply:
Added all four.